lib/elixir/lib/inspect/algebra.ex

```diff
@@ -105,6 +105,50 @@ defmodule Inspect.Opts do
           syntax_colors: [{color_key, IO.ANSI.ansidata()}],
           width: non_neg_integer | :infinity
         }
+
+  @doc """
+  Builds an `Inspect.Opts` struct.
+  """
+  def new(opts) do
+    struct(%Inspect.Opts{inspect_fun: default_inspect_fun()}, opts)
+  end
+
+  @doc """
+  Returns the default inspect fun.
+  """
+  def default_inspect_fun do
+    :persistent_term.get({__MODULE__, :inspect_fun}, &Inspect.inspect/2)
+  end
+
+  @doc """
+  Sets the default inspect fun.
+
+  Set this option with care as it will change how all values
+  in the system are inspected. The main use of this functionality
+  is to provide an entry point to filter inspected values,
+  in order for entities to comply with rules and legislations
+  on data security and data privacy.
+
+  It is extremely discouraged for libraries to set their own
+  function as this must be controlled by applications.
+
+  The default is `Inspect.inspect/2`.
+
+  ## Examples
+
+      previous_fun = Inspect.Opts.default_inspect_fun()
+
+      Inspect.Opts.default_inspect_fun(fn
+        %{address: _} = map, opts ->
+          fun.(%{map | address: "[REDACTED]"}, opts)
+
+        value, opts ->
+          fun.(map, opts)
+      end)
```

Comment:
```suggestion
      previous_fun = Inspect.Opts.default_inspect_fun()

      Inspect.Opts.default_inspect_fun(fn
        %{address: _} = map, opts ->
          previous_fun.(%{map | address: "[REDACTED]"}, opts)

        map, opts ->
          previous_fun.(map, opts)
      end)

```

Reply:
The other clause will match any value, no only maps.

Reply:
Updated